Scripts/Editor/Compiler/ScriptDefinesToggle.cs

```diff
@@ -0,0 +1,86 @@
+using System.Collections.Generic;
+using System.Linq;
+using UnityEditor;
+using UnityEditor.Build;
+
+namespace Anvil.Unity.DOTS.Editor.Compiler
+{
+    /// <summary>
+    /// Provides a quick and easy way to enable/disable commonly used Scripting Defines for
+    /// development and debugging. 
+    /// </summary>
+    [InitializeOnLoad]
+    public static class ScriptDefinesToggle
+    {
+        private const string MENU_PATH_BASE = "Anvil/Script Defines";
+        private const string DEFINE_ANVIL_DEBUG_SAFETY_EXPENSIVE = "ANVIL_DEBUG_SAFETY_EXPENSIVE";
+        private const string MENU_PATH_ANVIL_DEBUG_SAFETY_EXPENSIVE = MENU_PATH_BASE + "/" + DEFINE_ANVIL_DEBUG_SAFETY_EXPENSIVE;
+        private const string DEFINE_ANVIL_DEBUG_LOGGING_EXPENSIVE = "ANVIL_DEBUG_LOGGING_EXPENSIVE";
+        private const string MENU_PATH_ANVIL_DEBUG_LOGGING_EXPENSIVE= MENU_PATH_BASE + "/" + DEFINE_ANVIL_DEBUG_LOGGING_EXPENSIVE;
+
+        private static readonly HashSet<string> ACTIVE_DEFINES = new HashSet<string>();
```

Comment:
Can probably remove this. Just fetch the define symbols on each validator or set call.
Caching active defines here adds unnecessary complexity to this script.

Working directly with the values returned by `PlayerSettings` means there's a single point of truth.